Terminal.Gui/Windows/DefaultFileOperations.cs

```diff
@@ -0,0 +1,144 @@
+
+			int result = MessageBox.Query (
+				string.Format (Strings.fdDeleteTitle, adjective),
+				string.Format (Strings.fdDeleteBody, adjective),
+				Strings.fdYes, Strings.fdNo);
+
+			try {
+				if (result == 0) {
+					if (d is FileInfo) {
+						d.Delete ();
+					} else {
+						((DirectoryInfo)d).Delete (true);
+					}
+
+                    return true;
+				}
+			} catch (Exception ex) {
+				MessageBox.ErrorQuery (Strings.fdDeleteFailedTitle, ex.Message, "Ok");
+			}
+
+            return false;
+		}
+
+        private bool Prompt(string title,string defaultText, out string result)
+        {
+
+            bool confirm = false;
+            var btnOk = new Button("Ok"){
+                IsDefault = true,
+            };
+            btnOk.Clicked+=(s,e)=>{
+                confirm = true;
+                Application.RequestStop();
+            };
+            var btnCancel = new Button("Cancel");
+            btnCancel.Clicked+=(s,e)=>{
+                confirm = false;
+                Application.RequestStop();
+            };
+
+            var lbl = new Label(Strings.fdRenamePrompt);
+            var tf = new TextField(defaultText){
+                X = Pos.Right(lbl),
+                Width = Dim.Fill(),
+            };
+            tf.SelectAll();
+
+            var dlg = new Dialog(title){
+                Width = Dim.Percent(50),
+                Height = 4
+            };
+            dlg.Add(lbl);
+            dlg.Add(tf);
+
+            // Add buttons last so tab order is friendly
+            // and TextField gets focus
+            dlg.AddButton(btnOk);
+            dlg.AddButton(btnCancel);
+
+            Application.Run(dlg);
+
+            result = tf.Text?.ToString();
+
+            return confirm;
+        }
+
+		/// <inheritdoc/>
+		public FileSystemInfo Rename (FileSystemInfo toRename)
+		{
+            // Dont allow renaming C: or D: or / (on linux) etc
+            if(toRename is DirectoryInfo dir && dir.Parent == null)
+            {
+                return null;
+            }
+            
+            if(Prompt(Strings.fdRenameTitle,toRename.Name,out var newName))
+            {
+                if(!string.IsNullOrWhiteSpace(newName))
+                {
+                    try {
+
+                            if (toRename is FileInfo f) {
+                                
+                                var newLocation = new FileInfo(Path.Combine(f.Directory.FullName,newName));
+                                f.MoveTo(newLocation.FullName);
+                                return newLocation;
+
+                            } else {
+                                var d = ((DirectoryInfo)toRename);
+
+                                var newLocation = new DirectoryInfo(Path.Combine(d.Parent.FullName,newName));
+                                d.MoveTo(newLocation.FullName);
+                                return newLocation;
+                            }
+                    } catch (Exception ex) {
```

Comment:
The following Exceptions are possible here (Also `ArgumentNullException` and `ArgumentException` but those should be prevented by `IsNullOrWhiteSpace` check).  I generally just use `Exception` for file IO since  there are so many different kinds of excceptions.

[IOException](https://learn.microsoft.com/en-us/dotnet/api/system.io.ioexception?view=net-7.0)

An attempt was made to move a directory to a different volume.

-or-

destDirName already exists.

-or-

You are not authorized to access this path.

-or-

The directory being moved and the destination directory have the same name.
[SecurityException](https://learn.microsoft.com/en-us/dotnet/api/system.security.securityexception?view=net-7.0)

The caller does not have the required permission.
[DirectoryNotFoundException](https://learn.microsoft.com/en-us/dotnet/api/system.io.directorynotfoundexception?view=net-7.0)

The destination directory cannot be found.